### PR TITLE
Callback url support

### PR DESCRIPTION
**Pull request by mayankamencherla**
Added callback url support for opencart 2.3

### CODE REVIEW

File: catalog/view/theme/default/template/payment/razorpay.tpl [hunk 1]
```diff
@@ -15,6 +15,7 @@
     notes: {
       opencart_order_id: "<?php echo $merchant_order_id; ?>"
     },
+    callback_url: "<?php echo $return_url; ?>",
```
Comment (captn3m0) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: @pronav want to confirm the priority order for `handler` and `callback_url` being present.

**Comment by mayankamencherla** [comment]
@captn3m0 Does the PR look fine apart from this?

**Comment by captn3m0** [comment]
@mayankamencherla I'm not sure if we need the field support either, considering that we are enforcing checkout here. Have you been able to get the field validation error messages while testing?

File: catalog/view/theme/default/template/payment/razorpay.tpl (hunk 1 above)
Comment (pronav) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: It works as expected, handler > callback_url

File: catalog/controller/extension/payment/razorpay.php [hunk 2]
```diff
@@ -123,7 +123,16 @@ public function callback()
         } 
         else 
         {
-            echo 'An error occured. Contact site administrator, please!';
+            $error = $_POST['error'];
```
Comment (captn3m0) [inline catalog/controller/extension/payment/razorpay.php]: @mayankamencherla Have an isset here.

File: catalog/view/theme/default/template/payment/razorpay.tpl (hunk 1 above)
Comment (mayankamencherla) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: cool

File: catalog/controller/extension/payment/razorpay.php (hunk 2 above)
Comment (mayankamencherla) [inline catalog/controller/extension/payment/razorpay.php]: added

**Pull request merged**